daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -746,9 +754,13 @@ object Speedy {
   }
 
   /** The builtin arguments have been evaluated. Now execute the builtin. */
-  final case class KBuiltin(builtin: SBuiltin, actuals: util.ArrayList[SValue]) extends Kont {
+  final case class KBuiltin(builtin: SBuiltin, actuals: util.ArrayList[SValue], envSize: Int)
+      extends Kont {
     def execute(v: SValue, machine: Machine) = {
       actuals.add(v)
+      // we only need to call restoreEnvStack (not the full restoreEnv), because actuals
+      // are changed to evaluate the builtin, and a builtin has no free-vars.
+      machine.restoreEnvStack(envSize)
       // A builtin has no free-vars, so we dont have to set the frame.
       machine.actuals = actuals
```

Comment:
If we don't point into the frame, we can reset it and release the values in the old frame to the GC, can't we?
```suggestion
      machine.restoreEnv(null, actuals, envSize)
```

Reply:
I had wondered if it should be set to `null`. And of course it should for GC reasons. Thanks.